Parallel docker builds result in duplicate name error

### What happened?

My release process builds two container images. One for amd64 and one for arm64. These images are pushed with the `docker_manifests` feature in order to build a multi arch image.

I get this error most of the time. It seems to be more frequent on my AMD 5950x (32 thread) system. I also get this error on my Macbook pro but less frequently. It seems more common on the more powerful system.

```
   • docker images    
      • building docker image     image=myaccount/myapp-arm64:latest
      • building docker image     image=myaccount/myapp-amd64:latest
   ⨯ release failed after 80.69s error=docker build failed: failed to build myaccount/myapp-amd64:latest: exit status 1: #1 [internal] booting buildkit
#1 pulling image moby/buildkit:buildx-stable-1
#1 pulling image moby/buildkit:buildx-stable-1 3.4s done
#1 creating container buildx_buildkit_eloquent_lederberg0 done
#1 ERROR: Error response from daemon: Conflict. The container name "/buildx_buildkit_eloquent_lederberg0" is already in use by container "e3fb3308b53c663e6ebb3ee247d685aef4c6cf4955e63dd47a0e2e8539f7235f". You have to remove (or rename) that container to be able to reuse that name.
------
 > [internal] booting buildkit:
------
error: Error response from daemon: Conflict. The container name "/buildx_buildkit_eloquent_lederberg0" is already in use by container "e3fb3308b53c663e6ebb3ee247d685aef4c6cf4955e63dd47a0e2e8539f7235f". You have to remove (or rename) that container to be able to reuse that name.

Learn more at https://goreleaser.com/errors/docker-build

make: *** [Makefile:268: release-test] Error 1
```

### How can we reproduce this?

It feels like a race condition. It might play nice on your system, or it might not. Setup multiple docker builds and try to run the builld until it happens.

### goreleaser version

```bash
goreleaser version 1.9.1
commit: e8307e88345f36a84a3a1bb5c24d947ad68cce40
built at: 2022-05-20T14:44:16Z
built by: goreleaser
goos: linux
goarch: amd64
module version: v1.9.1, checksum: h1:YtQg3W9MFYinzRPuWQ1uAV0ikLz49JKBgAnZWsXk7Q4=
```
```


### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

I can work around this issue by setting parallel to 1, but this is very slow as I am building 13 binaries, 13 archives, 15 packages, and two container images. All working wonderfully outside of this :+1:

--- a/internal/pipe/docker/api_docker.go
+++ b/internal/pipe/docker/api_docker.go
@@ -61,7 +61,7 @@ func (i dockerImager) Build(ctx *context.Context, root string, images, flags []s
 func (i dockerImager) buildCommand(images, flags []string) []string {
 	base := []string{"build", "."}
 	if i.buildx {
-		base = []string{"buildx", "build", ".", "--load"}
+		base = []string{"buildx", "--builder", "default", "build", ".", "--load"}
 	}
 	for _, image := range images {
 		base = append(base, "-t", image)

--- a/internal/pipe/docker/docker_test.go
+++ b/internal/pipe/docker/docker_test.go
@@ -1122,7 +1122,7 @@ func TestBuildCommand(t *testing.T) {
 			name:   "buildx",
 			buildx: true,
 			flags:  []string{"--label=foo", "--build-arg=bar=baz"},
-			expect: []string{"buildx", "build", ".", "--load", "-t", images[0], "-t", images[1], "--label=foo", "--build-arg=bar=baz"},
+			expect: []string{"buildx", "--builder", "default", "build", ".", "--load", "-t", images[0], "-t", images[1], "--label=foo", "--build-arg=bar=baz"},
 		},
 	}
 	for _, tt := range tests {

--- a/www/docs/customization/docker.md
+++ b/www/docs/customization/docker.md
@@ -240,6 +240,27 @@ docker build -t myuser/myimage . \
 !!! tip
     Learn more about the [name template engine](/customization/templates/).
 
+## Use a specific builder with Docker buildx
+
+If `buildx` is enabled, the `default` context builder will be used when building
+the image. This builder is always available and backed by BuildKit in the
+Docker engine. If you want to use a different builder, you can specify it using
+the `build_flag_templates` field:
+
+```yaml
+# .goreleaser.yaml
+dockers:
+  -
+    image_templates:
+    - "myuser/myimage"
+    use: buildx
+    build_flag_templates:
+    - "--builder=mybuilder"
+```
+
+!!! tip
+    Learn more about the [buildx builder instances](https://docs.docker.com/buildx/working-with-buildx/#work-with-builder-instances).
+
 ## Podman
 
 !!! success "GoReleaser Pro"

